### PR TITLE
[SPARKR-156] phase 1: implement zipWithUniqueId() of the RDD class.

### PR DESCRIPTION
**Pull request by sun-rui**

### CODE REVIEW

**Comment by davies** [comment]
LGTM

File: pkg/R/RDD.R [hunk 1]
```diff
@@ -1427,6 +1427,47 @@ setMethod("setName",
...
+#' @aliases zipWithUniqueId,RDD
+setMethod("zipWithUniqueId",
+          signature(x = "RDD"),
+          function(x) {
+            n <- numPartitions(x)
+
+            partitionFunc <- function(split, part) {
+              index <- 0
+              result <- vector("list", length(part))
+              for (item in part) {
```
Comment (shivaram) [inline pkg/R/RDD.R]: It might be better to use `sapply` or `lapply` here instead of `for`

**Comment by sun-rui** [comment]
According to the comment, I use mapply() instead of for statement. Actually, I did think about using lapply() or mapply(), but had a concern that involving a sequence vector may consume more memory than using a for statement. Not sure if my concern makes sense.

**Comment by shivaram** [comment]
Thanks @sun-rui. LGTM, Merging this

**Pull request merged**